taichi/transforms/constant_fold.cpp

```diff
@@ -114,13 +100,11 @@ class ConstantFold : public BasicStmtVisitor {
                       rhs.dt,
                       true};
     auto *ker = get_jit_evaluator_kernel(id);
-    auto &current_program = stmt->get_kernel()->program;
-    // save input args to prevent the current kernel from being overridden.
-    ContextArgSaveGuard _(current_program.get_context());
     auto launch_ctx = ker->make_launch_context();
     launch_ctx.set_arg_raw(0, lhs.val_u64);
     launch_ctx.set_arg_raw(1, rhs.val_u64);
     (*ker)(launch_ctx);
+    auto &current_program = stmt->get_kernel()->program;
     ret.val_i64 = current_program.fetch_result<int64_t>(0);
```

Comment:
My future hope is:
```suggestion
    ret.val_i64 = launch_ctx.get_arg<int64_t>(0);
```

Reply:
I see your point. Unfortunately, reading the result is, in most cases, separated from calling the kernel.. I guess a preliminary step toward this goal would be to stop using `Program` to fetch the result.